# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.cpp
# Commit: 7333d4f4c7d1
# Full Hash: 7333d4f4c7d1b5ad585d222157067aa4276f9945
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-09-10 03:29:56
# Description:
#   Bug 1579437 - Check for discarded BrowsingContext in more places, r=farre
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45088
# ==============================================================================

diff -r 941fe2efca5d -r 7333d4f4c7d1 docshell/base/BrowsingContext.cpp
--- a/docshell/base/BrowsingContext.cpp	Mon Sep 09 17:30:08 2019 +0000
+++ b/docshell/base/BrowsingContext.cpp	Mon Sep 09 12:04:13 2019 +0000
@@ -945,7 +945,12 @@
                  aSubjectPrincipal, aError);
 }
 
-void BrowsingContext::Transaction::Commit(BrowsingContext* aBrowsingContext) {
+nsresult BrowsingContext::Transaction::Commit(
+    BrowsingContext* aBrowsingContext) {
+  if (NS_WARN_IF(aBrowsingContext->IsDiscarded())) {
+    return NS_ERROR_FAILURE;
+  }
+
   if (!Validate(aBrowsingContext, nullptr)) {
     MOZ_CRASH("Cannot commit invalid BrowsingContext transaction");
   }
@@ -973,7 +978,9 @@
   }
 
   Apply(aBrowsingContext);
+  return NS_OK;
 }
+
 bool BrowsingContext::Transaction::Validate(BrowsingContext* aBrowsingContext,
                                             ContentParent* aSource) {
 #define MOZ_BC_FIELD(name, ...)                                        \